### DISCUSSION TITLE
BRouter: viaX-position not mapped

### DISCUSSION THREAD

**Discussion by WolfgangZ42** [devemux86/cruiser#170] [Q&A] [answered]
**A strange behavior of BRouter.**
First my BRouter mapping - updated today:
<img width="400" src="https://github.com/devemux86/cruiser/assets/121806238/8d6fea6e-3f08-40c6-b85a-622b761aae5a"/>

Import this file into Cruiser: [F_1-a1b.zip](https://github.com/devemux86/cruiser/files/11504563/F_1-a1b.zip)  
Moving any waypoint (without actually moving it) then results in:

<img width="400" src="https://github.com/devemux86/cruiser/assets/121806238/1da3fb98-7e72-4330-b7a6-cbeedbb2326b"/>
Routing-Service = **BRouter(offline)**
<img width="400" src="https://github.com/devemux86/cruiser/assets/121806238/c0d55cd5-caed-475b-8c55-e4171eecf7e7"/>
Routing-Service = **BRouter** 

Any Idea?

**Comment by devemux86** [comment]
@WolfgangZ42 thanks for the report.

> Moving any waypoint (without actually moving it)

This is a nice trick that forces the app to recalculate the route!
So it tries to recalculate it with the selected profile + service + settings.

As we discussed in https://github.com/devemux86/cruiser/discussions/169, the message from BRouter (offline):
"BRouter: via3-position not mapped in existing datafile"
means that the mentioned waypoint cannot be attached to the nearest road.

In case of BRouter website the result seems the same.

What routing profile and settings do you use?
Have you tried to zoom in and move slightly the waypoint?

**Reply by WolfgangZ42** [reply]
> Have you tried to zoom in and move slightly the waypoint?

Sure I did that with waypoint 3 - unsuccesfully.

**Reply by WolfgangZ42** [reply]
My pref: [#T.zip](https://github.com/devemux86/cruiser/files/11505550/T.zip)

**Comment by 0709wiwiwi** [comment]
"motor" or "car".brf  does not correspond with

highway track
track type grade3
https://www.openstreetmap.org/way/129757725 

It needs this extra:
motor_vehicle yes
motor vehicle yes
motorbike yes
https://www.openstreetmap.org/way/40109540#map=15/46.4967/12.7962

**Reply by WolfgangZ42** [reply]
Willy,  both of the OSM parts you mentioned are accepted in the whole route: [Christians_Alpentour.zip](https://github.com/devemux86/cruiser/files/11505328/Christians_Alpentour.zip)
My example above is just an extract of this route. 

The whole story is: 
- Christian gave me his route with over 100 waypoints and the BRouter message 
- He wanted two parts bridged by direct line
- I erased unnecessary waypoints in kurviger.de to ~20 to save the enormous calculation time
- Then I inserted the "beeline" parts
- After that the BRouter message was gone 
- But still there even in parts of the original route 

My (or better: Christians) Problem is solved - but I thought it's worth investigating. My pref: [#T.zip](https://github.com/devemux86/cruiser/files/11505520/T.zip)

**Reply by 0709wiwiwi** [reply]
It will possibly be more comfortable by multi profiles per track section.

Except for a beeline  section currently a default profile applies to the entire track  :-(( 

You could then probably overcome difficult parts in a traject without "must" use beeline(s). 

Provided, of course, that a path is already drawn in osm.

The Zossebart reroute profile that I am just testing is attached  
[reroute.brf.txt](https://github.com/devemux86/cruiser/files/11505733/reroute.brf.txt)
I have assigned it in the Brouter app to Foot shortest.

**Comment by devemux86** [comment]
I asked @afischerdev and as always he is very helpful! :+1: 

There is the profile parameter `waypointCatchingRange` with default `250`.

**Reply by devemux86** [reply]
Next Cruiser will allow to change the BRouter `waypointCatchingRange` parameter inside Cruiser settings.
Local tests with a range > 250 m show successful routes with waypoints far from the road network.

What maximum do you think we can have, 1 km or more?

**Comment by devemux86** [comment]
Next Cruiser will also allow to change the BRouter `correctMisplacedViaPointsDistance` parameter (to avoid u-turns) in Cruiser settings.

What maximum do you think we can have, 100 m or more?

<img src="https://github.com/devemux86/cruiser/assets/3484020/3a496d48-5576-491a-8691-09220b691420" width="50%"/>

**Reply by 0709wiwiwi** [reply]
> correctMisplacedViaPointsDistance

100m max ok.
@WolfgangZ42  ok?

**Reply by WolfgangZ42** [reply]
I think my (or better: Christian's) problem had nothing to do with this distance. Rather, kurviger.de has allowed a route that is rejected by BRouter when recalculating - until now I thought Kurviger was more restrictive. Should the situation arise again, we will work consequently without recalculation.
So the default of 250 m for WaypointCatchingRange seems to me to be sufficient.

**Reply by devemux86** [reply]
@WolfgangZ42 your report seems indeed like a tags issue.

However it is convenient to have a more flexible range for waypoints.
It happens to place waypoints far from the road network and BRouter then complains.

**Reply by WolfgangZ42** [reply]
Short supplement: the route [Sauris.gpx.txt](https://github.com/devemux86/cruiser/files/11643300/Sauris.gpx.txt) works if **_use_tracks_** is enabled in options. After disabling it the error message appears again _"... viaxx-position not mapped ..."_
Hint: if you ride on gravel roads don't forget to switch on **_use_tracks_** ❗

**Reply by 0709wiwiwi** [reply]
Nice example of a gpx navigation track. By avg speed 70.1 kmh.
Anyway I noticed that at Shaping Point 11 the U-turn is missing  
When I recalculate with the Brouter offline the U-turn is attached.

**Comment by devemux86** [comment, marked as answer]
BRouter settings for "Waypoints search range" and "Correct misplaced waypoints" are available in [Cruiser 3.0.20](https://play.google.com/store/apps/details?id=gr.talent.cruiser).

Available in "Settings | Routing | BRouter | Advanced" or search settings for "brouter".

**Comment by Cer-o** [comment]
[Murcia - Mojácar.gpx.zip](https://github.com/devemux86/cruiser/files/11645175/Murcia.-.Mojacar.gpx.zip)
Good morning. I have also suffered this problem when importing different routes, such as the one I attached.
After reading the thread, the cause and possible solution to the problem have not been clear to me.

**Reply by devemux86** [reply]
@Cer-o 

If it is a Kurviger route, you can import the .kurviger file, offline without route calculation.

Importing a regular GPX file requires route calculation.
I can import your route without problem with the default BRouter options:

<img src="https://github.com/devemux86/cruiser/assets/3484020/c7327c88-0024-4d3a-b8b9-97252242c636" width="50%"/>

**Reply by Cer-o** [reply]
I think I have not understood your answer, still I have passed the path to kurviger format and imported it to Cruiser successfully.
I don't understand why it doesn't work importing the .gpx and if it does with the .kurviger.

**Reply by devemux86** [reply]
@Cer-o 

> I don't understand why it doesn't work importing the .gpx

I tried your gpx file with BRouter (offline) and it is imported successfully as the route displayed above.

Do you have the latest Cruiser?
What routing settings do you use in the app? Can you export the settings as .pref file?
What options do you use in import dialog?

**Reply by Cer-o** [reply]
[configuración CRUISER.pref.zip](https://github.com/devemux86/cruiser/files/11645504/configuracion.CRUISER.pref.zip)

I have the latest version of Cruiser (3.0.22)
In this route I get an error in via48
[Lario - Pinto.gpx.zip](https://github.com/devemux86/cruiser/files/11645506/Lario.-.Pinto.gpx.zip)


![image](https://github.com/devemux86/cruiser/assets/122309655/382f7fc0-a86a-488c-9ef8-8007ea40a86c) 
![image](https://github.com/devemux86/cruiser/assets/122309655/5b2fb23e-c039-4b28-ad58-7a9983ef70a2)

**Reply by 0709wiwiwi** [reply]
The Presented gpx track is a recorded simple track. At some places there where recorded some out and returns. This recorded simple track does not contain turninstructions. So you have to clean and reconstruct a navigation track and thereby display the recorded simple track as visual reference tot fine tune your finale design.
If you like the result you can save  as a gpx navtrack. At the gpx export set inclusive navigation.
At a later gox navtrack import select automatic.

An importable gpx navtrack is attached. 
[Cero_20230604_134039.gpx.txt](https://github.com/devemux86/cruiser/files/11645513/Cero_20230604_134039.gpx.txt)

Remove the .txt

**Reply by devemux86** [reply]
@Cer-o thanks for the extra details.

The message "BRouter: via48-position not mapped" means that this waypoint cannot be placed on the road network.
You can either move the waypoint closer to the road network
or increase the "Settings | Routes | BRouter (offline) | Advanced | Waypoints search range".

For example, if we set it to 500m then the import works:
(perhaps we need to set a larger value as default)

<img src="https://github.com/devemux86/cruiser/assets/3484020/14fb89fb-74cc-4097-9046-c783208afbe0" width="50%"/><img src="https://github.com/devemux86/cruiser/assets/3484020/7d1addd2-a6c9-49a1-b998-bb2d20a307c5" width="50%"/>

**Reply by Cer-o** [reply]
I have repositioned point 48 in the center of the road and the error has been reproduced again.
I have increased the setting you have indicated to me to 500 and the error has disappeared.
Thank you so much.

**Reply by devemux86** [reply]
@Cer-o 

> I have repositioned point 48 in the center of the road and the error has been reproduced again.

There is probably a barrier / gate / seasonal closure on the road of point 48?

You can enable the following BRouter setting and try again:

<img src="https://github.com/devemux86/cruiser/assets/3484020/1ce91207-d8f4-411a-836f-8e6644f30f44" width="50%"/>

**Reply by Cer-o** [reply]
I have reverted the distance to 300 meters and checked the ignore barriers, but the error has reproduced.
I have redialed the distance of 500 meters and the error has disappeared.

**Reply by Cer-o** [reply]
> La pista gpx presentada es una pista simple grabada. En algunos lugares allí donde se grabaron algunas salidas y vueltas. Esta pista simple grabada no contiene instrucciones de giro. Por lo tanto, debe limpiar y reconstruir una pista de navegación y, por lo tanto, mostrar la pista simple grabada como referencia visual para afinar su diseño final.Si te gusta el resultado puedes guardarlo como un gpx navtrack. En el conjunto de exportación gpx, la navegación inclusiva.En una importación posterior de gox navtrack, seleccione automático.
> 
> Se adjunta un gpx navtrack importable. [Cero_20230604_134039.gpx.txt](https://github.com/devemux86/cruiser/files/11645513/Cero_20230604_134039.gpx.txt)
> 
> Quitar el .txt

Hello and thanks for your help.
Due to my limited knowledge and the translation, I cannot understand you.

**Reply by 0709wiwiwi** [reply]
Ok nuevo intento.
1. El track gpx que has mostrado es un simple track record gpx.
2. Un track gpx simple no contiene instrucciones de navegación.
3. El track grabado contiene errores en los que te metes en una carretera falsa, que luego se corrigen con un retorno.
4. El track probablemente también indica algunas rutas a miradores seguidas de un retorno.

Utilice este track grabado como ejemplo visual para diseñar un NUEVO track de navegación. 
Cruiser puede hacer esto parcialmente de forma automática para usted, pero no 100% perfecto. 
Por lo que debe comprobar el resultado de algunos errores y corregirlos manualmente.

Una vez que el NUEVO diseño esté a tu gusto, puedes usarlo para navegar. 
O puedes exportarlo como un archivo .kurviger o archivo navtrack gpx para su posterior reutilización.

Traducción realizada con la versión gratuita del traductor www.DeepL.com/Translator

**Comment by 0709wiwiwi** [comment]
> For example, if we set it to 500m then the import works:  (perhaps we need to set a larger value as default)

A higher tolerance for motorcyclists could potentially be practical. 
For pedestrian routing, you would better check why a position has been refused.

Note: 
Presets can perfectly take care of such individual transportation or vehicle distinction rules.

**Reply by devemux86** [reply]
Important is that all other routers operate with large waypoint search ranges.
BRouter seems to be more restrictive and this is causing problems to users.

We definitely need to think about the user convenience with a larger range.
